Documentation/NvimConfiguration.md

```diff
@@ -0,0 +1,180 @@
+# NVim Project Configuration
+
+NVim can be configured to use the [COC-clangd](https://github.com/clangd/coc-clangd)
+plugin to provide code-completion as well as inline
+[git blame](https://github.com/f-person/git-blame.nvim) using [vim-plug](https://github.com/junegunn/vim-plug).
+
+Make sure you ran `./Meta/serenity.sh rebuild-toolchain` as well as
+`./Meta/serenity.sh run`.
+
+# Install vim-plug
+
+See [https://github.com/junegunn/vim-plug](https://github.com/junegunn/vim-plug).
+
+# Install coc.nvim
+
+The config file for neovim is at `~/.config/nvim/init.vim` or if
+set `$XDG_CONFIG_HOME/nvim/init.vim`.
+
+Add the Plugin:
+
+```vim
+Plug 'neoclide/coc.nvim', { 'branch': 'release' }
+```
+
+Run `:PlugInstall` inside nvim.
+
+# Install coc-clangd via CocInstall
+
+```vim
+:CocInstall coc-clangd
+```
+
+Note: This guide is tested with clangd version 14.0.6 and 15.0.6.
+
+In case you have not installed clangd already install it with
+`:CocCommand clangd.install`.
+
+This will install a separate version of clangd just for neovim.
+
+# Configure coc-clangd in coc-settings.json
+
+Use the following settings to ensure that coc-clangd works out of the box.
+
+Note: You might want to adjust the `clangd.fallbackFlags` depending on your build
+system and customize the `inlayHints.sep` based on your preference.
+
+```json
+{
+    "clangd.fallbackFlags": ["-std=c++20"],
+    "semanticTokens.enable": true,
+    "inlayHint.subSeparator": "︴",
+    "inlayHints.enableParameter": true,
+    "clangd.inlayHints.sep": "⇝"
+}
+```
+
+To change the coc-settings.json go to the file `~/.config/nvim/coc-settings.json`
+or type `:CocConfig` in the command line.
+
+Note: In case you already had another c++ language server configured in the
+`coc-settings.json` you might want to nuke it first and
+work towards your desired config by adding the other parts back in to avoid
+conflicts.
+
+**Don't have a languageServer config for clangd**. Allegedly (according to
```

Comment:
This is still badly formatted.

Reply:
Now i just copied the literal words of of coc-clangd. Sorry for the inconvenience.